pkg/platform/model/recipe.go

```diff
@@ -144,6 +145,8 @@ func fetchRecipeID(commitID strfmt.UUID, owner, project, orgID string, private b
 	var err error
 	params := iop.NewSolveOrderParams()
 	params.SetHTTPClient(retryhttp.DefaultClient.StandardClient())
+	timeout := params.HTTPClient.Timeout * time.Duration(retryhttp.DefaultClient.RetryMax)
```

Comment:
Can you be sure `params.HTTPClient.Timeout` isn't `0` here?

And what's the rationale for multiplying it by `RetryMax`?

Makes sense applying this for other similar cases.

Reply:
It's safe to assume here because NewSolveOrderParams uses a global default value, but it's probably not safe to assume in most Params setups. The rationale for multiplying it is that if we limit downloads to, say 30 seconds, we would potentially need to accommodate a total duration of {retry_count} x 30 seconds to prevent premature cancellation.

I consider this a fairly vulgar solution, and it's not appearing to work without more digging into how the various timeouts and context deadlines work together. This is very much related to my concerns about http client resource reuse. Parallel to that, we have this issue in a number of places in different forms.

Reply:
Can you just add some simple logic to verify it's not 0, should be really cheap.

 Also I don't understand the rationale for multiplying. What do downloads have to do with this? This is for solving an order.

Reply:
Sure! As for multiplying, if that isn't done, then it could happen that the 5th try works after 29 seconds, then has only 1 second to complete. By extending it to the retryCt*timeout, we can be sure to give a fair amount of time during the connection once it's made. I've no strong feelings about it.

Reply:
Let's stick to just retrying for now. I see what you mean but we're not doing this anywhere else yet. Would prefer if we consider that to be an important aspect that we roll it out universally. Additionally still not sold on using RetryMax to multiply..